ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,529 @@
+                "\n            AddFilter(FilterName, FilterValue);",
+                "\n        if(FilterAction == \"Remove\")",
+                "\n            RemoveFilter(FilterName, FilterValue);",
+                "\n        return InputDataFilter;",
+                "\n    }",
+                "\n",
+                "\n    private void AddFilter(string filterName, string filterValue)",
+                "\n    {",
+                "\n        if(InputDataFilter == null)",
+                "\n            InputDataFilter = Enumerable.Empty<(string, string)>().ToArray();",
+                "\n        if(!InputDataFilter.Contains((filterName, filterValue)))",
+                "\n            InputDataFilter = InputDataFilter.Append((filterName, filterValue)).ToArray();",
+                "\n    }",
+                "\n    ",
+                "\n    private void RemoveFilter(string filterName, string filterValue)",
+                "\n    {",
+                "\n        if(InputDataFilter != null && InputDataFilter.Contains((filterName, filterValue)))",
+                "\n        {   var f = InputDataFilter.ToList();",
+                "\n            f.Remove((filterName, filterValue));",
+                "\n            InputDataFilter = f.ToArray();",
+                "\n        }",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Slice and Dice"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface BasicSliceAndDiceFormsEntity : MutableScopeWithWorkspace {",
+                "\n    [DropdownValues(\"\", \"GroupOfContract\")]",
+                "\n    string SliceRowName { get; set; }",
+                "\n",
+                "\n    [DropdownMethod(nameof(GetSliceColumnNameAutocomplete))]",
+                "\n    string SliceColumnName { get; set; }",
+                "\n",
+                "\n    protected IReadOnlyCollection<string> InputRowSlices => (SliceRowName is null ? Enumerable.Empty<string>() : SliceRowName.RepeatOnce()).ToArray();",
+                "\n    [NotVisible] IReadOnlyCollection<string> defaultRowSlices { get; set; }",
+                "\n    protected string[] rowSlices => defaultRowSlices.Union(InputRowSlices).ToArray();",
+                "\n",
+                "\n    protected IReadOnlyCollection<string> InputColumnSlices => (SliceColumnName is null ? Enumerable.Empty<string>() : SliceColumnName.RepeatOnce()).ToArray();",
+                "\n    [NotVisible] IReadOnlyCollection<string> defaultColumnSlices { get; set; }",
+                "\n    protected string[] columnSlices => defaultColumnSlices.Union(InputColumnSlices).ToArray();",
```

Comment:
The order of the column slices is important to have a good experience. 
How can one control the order given that there are default slices which might need to be moved after the input slices?

Reply:
The slice control is not final, and probably won't be before the PR is good enough for merging. The snd control should have a different format including the feature you are looking for.